tap_bigquery/sync_bigquery.py

```diff
@@ -207,11 +207,14 @@ def do_sync(config, state, stream):
                             day=row[key].day)
                     elif type(row[key]) == datetime.datetime:
                         r = row[key]
+                    elif row[key] is None:
```

Comment:
We should raise if `prop.type[0]` is not 'null' when row[key] is None.
So it can be something like:
```
elif row[key] is None:
    if prop.type[0] != "null":
        raise ValueError("NULL value not allowed by the schema")
    r = row[key]
```

It would be nice if you can do such null check for string, number, and integer types in this PR :)